src/components/icons/IconBadge.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import { Platform, StyleProp, View, ViewStyle } from 'react-native'
+
+import { useAccountSettings } from '../../actions/LocalSettingsActions'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+
+export interface Props {
+  children: React.ReactNode
+  /** Size of the child icon */
+  sizeRem: number
+
+  /** To adjust the position of the dot */
+  offsetX?: number
+  /** To adjust the position of the dot */
+  offsetY?: number
+}
+
+/**
+ * Maybe renders a red dot badge on top of the supplied `children,` with a white
+ * number or dot inside.
+ *
+ * For backwards compatibility, takes a style prop and provides no built-in margins.
+ */
+export const IconBadge = (props: Props) => {
+  const { children, offsetX = 0, offsetY = 0, sizeRem } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const { notifState } = useAccountSettings()
+  const priorityNotifs = Object.values(notifState).filter(notifInfo => notifInfo.isPriority && !notifInfo.isCompleted).length
+  const incompleteNotifs = Object.values(notifState).filter(notifInfo => !notifInfo.isCompleted).length
```

Comment:
We shouldn't be mixing business logic with display logic. Please pull this logic out to the notification scene / side menu button, and instead take a prop like `count?: number` here in the badge component.

If you want, you could create a `useNotificationCount` hook to return the number derived here, to make life easier in those other places.